haskell-indexer-pipeline-ghckythe-wrapper/src/Language/Haskell/Indexer/Args.hs

```diff
@@ -109,10 +121,21 @@ flagParser = Flags
              <> help ("Overrides the preprocessor binary, but keeps it's "
                       ++ "options. Note: other tools can still be overriden "
                       ++ "by passing the regular -pgmX GHC options.")))
+     <*> optional (strOption
+            ( long "output"
+            <> short 'o'
+            <> metavar "PATH"
+            <> help ("The location to write the indexes to")))
 
 index :: [String] -> Flags -> IO ()
-index args Flags{..} = do
+index args fs = do
     lock <- newMVar ()
+    indexX (ghcToKythe lock) args fs
+
+indexX ::
```

Comment:
I wonder about the X in the name.

Reply:
It's a convention from GHC where `indexX` is like `index` but with some additional abstraction. The X is meant to indicate an additional hole that be filled in.